reconcile/utils/terrascript_client.py

```diff
@@ -3991,5 +3979,73 @@ def populate_tf_resource_asg(self, resource, namespace_info):
         output_value = '${' + template_resource.latest_version + '}'
         tf_resources.append(Output(output_name_0_13, value=output_value))
 
-        for tf_resource in tf_resources:
-            self.add_resource(account, tf_resource)
+        self.add_resources(account, tf_resources)
+
+    def populate_tf_resource_route53_zone(self, resource, namespace_info):
+        account, identifier, common_values, output_prefix, \
+            output_resource_name, annotations = \
+            self.init_values(resource, namespace_info)
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info, output_prefix,
+                                 output_resource_name, annotations)
+
+        # https://www.terraform.io/docs/providers/aws/r/route53_zone.html
+        values = {
+            'name': identifier,
+            'tags': common_values['tags'],
+        }
+        zone_id = safe_resource_id(identifier)
+        zone_tf_resource = aws_route53_zone(zone_id, **values)
+        tf_resources.append(zone_tf_resource)
+
+        # outputs
+        # zone id
+        output_name_0_13 = output_prefix + '__zone_id'
+        output_value = f'${{{zone_tf_resource.zone_id}}}'
+        tf_resources.append(Output(output_name_0_13, value=output_value))
+        # name servers
+        output_name_0_13 = output_prefix + '__name_servers'
+        output_value = f'${{{zone_tf_resource.name_servers}}}'
+        tf_resources.append(Output(output_name_0_13, value=output_value))
+
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "route53:Change*",
+                        "route53:Create*",
```

Comment:
speaking about roles, seems like we're missing one?

`route53:DeleteHostedZone` or  `route53:Delete*` as what I see from https://github.com/openshift/hive/blob/master/docs/using-hive.md#managed-dns-1

Sorry for the noise if this is covered already :) great job :+1:

Reply:
thanks!

i did this intentionally so Hive doesn't accidentally delete the zone. it is created through app-interface, so i was also reluctant to give `Create`, but i don't mind if it attempts to create something that exists.

Reply:
How can hive accidentally delete a zone? I would assume that it could be only on a case where hive would intentionally delete a record under the hosted zone, are there other exceptions?
Speaking about giving create, I'm not sure either. We should be ok as long as we _update_/ _populate_ records under the hosted zone

Reply:
i don't know how hive can delete a zone, but we are not allowing that anyway :)

Reply:
sounds cool :) lgtm :+1: